### ISSUE TITLE
Regression: submodules of included OUT_DIR module are not properly resolved

### ISSUE THREAD

**Issue by TitanNano** [rust-lang/rust-analyzer#17699]
Since the most recent stable release 2024-07-22 symbols inside a submodule of a module which has been included from the `OUT_DIR` via `include!()` are no longer properly resolved.

I don't see any errors in my editor or the RA log, but autocompletion, documentation, and code navigation for symbols which have been imported from such a module no longer work. 

Attempting to navigate to a definition results in a "No definition found." error. It's possible to navigate into the included file, but not further inside.

---
**rust-analyzer version**: rust-analyzer 0.0.0 (4afe0d539 2024-07-21)

**rustc version**: rustc 1.79.0 (129f3b996 2024-06-10)

**editor or extension**: helix

**repository link (if public, optional)**: https://github.com/godot-rust/gdext

**code snippet to reproduce**: 
```rust
mod gen {
    include!(concat!(env!("OUT_DIR"), "/mod.rs"));
}
```

OUT_DIR/mod.rs
```rust
pub mod classes;
```

OUT_DIR/classes/mod.rs
```rust
pub struct TestStruct {}
```
---
Bisecting the issue shows that this is the first bad commit: https://github.com/rust-lang/rust-analyzer/commit/b9469f52a37dccf90dc5a76ad51b5f54c498206a

**Comment by lnicola** [comment]
Can you try a newer commit? This was probably fixed by #17660.

**Comment by TitanNano** [comment]
Yes, that commit fixes the issue. Apologies for the noise.
